core/src/main/java/io/substrait/expression/proto/ProtoExpressionConverter.java

```diff
@@ -109,7 +112,73 @@ public Expression from(io.substrait.proto.Expression expr) {
         yield ImmutableExpression.ScalarFunctionInvocation.builder()
             .addAllArguments(args)
             .declaration(declaration)
-            .outputType(protoTypeConverter.from(expr.getScalarFunction().getOutputType()))
+            .outputType(protoTypeConverter.from(scalarFunction.getOutputType()))
+            .build();
+      }
+      case WINDOW_FUNCTION -> {
+        var windowFunction = expr.getWindowFunction();
+        var functionReference = windowFunction.getFunctionReference();
+        SimpleExtension.WindowFunctionVariant functionVariant;
+        try {
+          functionVariant = lookup.getWindowFunction(functionReference, extensions);
+        } catch (RuntimeException e) {
+          // TODO: Ideally we shouldn't need to catch a RuntimeException to be able to attempt our
+          // second lookup
```

Comment:
Not as part of this PR, but I think it would be good to move substrait-java and isthmus away from throwing RuntimeExceptions and towards a hierarchy of checked exceptions to improve library economics.